Remove (potentially) recursive DEBUG statements

mjcarroll opened the pull request:
This is an alternative fix to #1241.   This also addresses ros/rosconsole#30

When using a log level of `debug`, every logging statement will have console output: 
```Warning: recursive print statement has occurred.  Throwing out recursive print.```

The logic introduced in #1241 uses `getCached` which will cause a second logging statement to occur recursively when log level is set to debug.

Edit: This has been updated to remove the offending DEBUG statements so that it will work in all cases.

mjcarroll commented:
Alternatively: this is the offending log statement: https://github.com/ros/ros_comm/blob/d487d5877049d61fe2ebcc6487e9a7a72d9f0346/clients/roscpp/src/libros/param.cpp#L299

Is it likely that people are debugging the cache in normal operation?  Is this something that could be removed?

dirk-thomas commented on clients/roscpp/src/libros/param.cpp:
```diff
@@ -472,12 +462,12 @@ bool getCached(const std::string& key, XmlRpc::XmlRpcValue& v)
 }
 
-template <class T> T xml_cast(XmlRpc::XmlRpcValue xml_value) 
+template <class T> T xml_cast(XmlRpc::XmlRpcValue xml_value)
```
Please revert this urelated whitespace change.

Same below.

mjcarroll replied:
Whitespace changes removed.

Commit:
Remove DEBUG statements from getImpl

These statements can potentially be called from within a logging
function, which causes a "recursive print" warning.

Signed-off-by: Michael Carroll <michael@openrobotics.org>

flixr commented:
These debug messages can be useful sometimes, but while I personally prefer #1597 this is also acceptable to me.
The current state makes debugging impossible though and we really need to fix this one way or another!

dirk-thomas commented:
@ros-pull-request-builder retest this please
